taf/git.py

```diff
@@ -62,18 +74,23 @@ def __init__(
 
     @classmethod
     def from_json_dict(cls, json_data):
-        path = json_data.pop("path")
-        urls = json_data.pop("urls")
-        custom = json_data.pop("custom", None)
-        name = json_data.pop("name")
-        default_branch = json_data.pop("default_branch", "master")
-        return cls(path, urls, custom, default_branch, name, **json_data)
+        """Create a new instance based on data contained by the `json_data` dictionary,
+        which can be create by calling `to_json_dict`
+        """
+        path = str(Path(json_data.get("path")))
+        name = json_data.get("name")
+        if name is not None:
+            name = str(Path(name))
+            if path.endswith(name):
+                json_data.pop("name")
+        return cls(**json_data)
 
     def to_json_dict(self):
+        """Returns a dictionary mapping all attributes to their values"""
         return {
```

Comment:
I'd like to keep that check and fully support creation of repositories given just its path. It does make sense to split the path here though so that the json data can easily be used to create a repository either manually or using `from_json`.